tools/c7n_azure/c7n_azure/actions.py

```diff
@@ -52,28 +52,60 @@ def update_resource_tags(tag_action, resource, tags):
             if tag_action.manager.type == 'armresource':
                 raise NotImplementedError('Cannot tag generic ARM resources.')
 
+            # build a GenericResource object from the resource string returned by the Azure API
             az_resource = GenericResource.deserialize(resource)
             api_version = tag_action.session.resource_api_version(az_resource.id)
+
+            # set the tags for the VM
             az_resource.tags = tags
```

Comment:
I suspect this isn't what we want - this API call just updates the fields you pass, so you only want to pass the tags.  By passing everything you are doing a full update of every field on the resource, which a) assumes you have a valid copy to start with and b) assumes no one else was editing it in the meantime.

I think you want something more similar to what happens above this code with Resource Groups - where you create a structure with only the tags and nothing else.  That code uses something called "ResourceGroupPatchable" and I'm not sure what that is :) But likely we just need to pass `{ tags=tags }` somehow and nothing else.

Reply:
The docs Laurent sent us in the email the parameters is the whole `GenericResource` like Aaron applied here. He said the update was the patchable one, although I'm still concerned he needed to set a service plan to None to get this to work.... https://docs.microsoft.com/en-us/python/api/azure-mgmt-resource/azure.mgmt.resource.resources.v2017_05_10.operations.ResourcesOperations?view=azure-python#update

Reply:
Well I think a generic resource just has a property bag so it can be mostly empty or well populated - I'd see what is ending up in this one and be sure to look at the ARM writes that are occuring and see that they are only writing tags.  Pretty sure this is writing everything.  Might have to clear out the property bag, or just make a net new generic resource and set the ID's and tags.

Reply:
@stefangordon @aluong FYI on latest commt. Thanks to @erwelch we get a newed up GenericResource object working just passing the tags property. Don't know what was doing last night that caused this not to work. Looks pretty clean now.